File: MyAnimeListJava/src/main/java/com/example/myanimelistjava/models/Anime.java
package com.example.myanimelistjava.models;

import lombok.Data;

import java.util.Date;
import java.util.List;
import java.util.UUID;

@Data
public class Anime {
    private String title;
    private String titleEnglish;
    private String types;
    private int episodes;
    private String status;
    private Date date;
    private String rating;
    private List<String> genres;
    private String img;
    private final UUID id= UUID.randomUUID();

    public Anime(String title, String title_english, String type, int episodes, String status, java.sql.Date releaseDate, String rating, List<String> genre, String imageUrl, UUID id) {
    }
}

File: MyAnimeList/src/main/kotlin/com/example/myanimelist/models/Anime.kt
package com.example.myanimelist.models

import java.time.LocalDate
import java.util.*

data class Anime(
    var title: String,
    var titleEnglish: String,
    var types: String,
    var episodes: Int,
    var status: String,
    var date: LocalDate,
    var rating: String,
    var genres: List<String>,
    var img: String?,
    val id: UUID = UUID.randomUUID()
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Anime

        if (id != other.id) return false

        return true
    }

    override fun hashCode(): Int {
        return id.hashCode()
    }
}

